crates/red_knot_python_semantic/src/types/string_annotation.rs

```diff
@@ -5,10 +5,10 @@ use ruff_python_ast::{self as ast, ModExpression, StringFlags};
 use ruff_python_parser::{parse_expression_range, Parsed};
 use ruff_text_size::Ranged;
 
-use crate::types::diagnostic::{TypeCheckDiagnostics, TypeCheckDiagnosticsBuilder};
+use crate::types::diagnostic::report_type_diagnostic;
 use crate::Db;
 
-type AnnotationParseResult = Result<Parsed<ModExpression>, TypeCheckDiagnostics>;
+type AnnotationParseResult = Result<Parsed<ModExpression>, ()>;
```

Comment:
Minor: Something like

```suggestion
struct StringAnnotationParseError;
type AnnotationParseResult = Result<Parsed<ModExpression>, StringAnnotationParseError>;
```

would maybe make `match`es at the call sites of this function a bit easier to understand